rest/server_context.go

```diff
@@ -159,6 +165,8 @@ func (sc *ServerContext) Close() {
 	sc.lock.Lock()
 	defer sc.lock.Unlock()
 
+	sc.ctxCancel()
```

Comment:
Do we need to wait for the bootstrap polling loop to terminate before we continue with server context close?  Potentially prior to acquiring the sc.lock?  It seems like there's a potential race where fetchAndLoadConfigs has fires in the polling goroutine after we acquire the lock.  So applyConfigs is blocked on acquire sc.lock, but then runs after we've finished Close() and possibly recreates the databases we just closed.  Unlikely to be a problem in production as sc.Close() doesn't get called other than during shutdown, but seems like the kind of thing that will cause integration testing headaches.

Reply:
Yep makes sense, thanks. I'll take a closer look at this close behaviour once I'm back.

Reply:
Reworked this to align with the stats process, and added a util for the `terminator` -> `done` channel pattern we're using for these syncronous closes.